code/modules/voidrunning/structures/netchair.dm

```diff
@@ -0,0 +1,173 @@
+
+/obj/structure/netchair/ui_act(action, params)
+	. = ..()
+	if(.)
+		return TRUE
+	switch(action)
+		if("select_outfit")
+			var/datum/outfit/new_suit = resolve_outfit(params["outfit"])
+			if(new_suit)
+				netsuit = new_suit
+				return TRUE
+
+	return FALSE
+
+/// Creates a new z-level, an avatar, then transfers the mind to the avatar.
+/obj/structure/netchair/proc/enter_matrix(mob/living/carbon/human/neo)
+	set waitfor = FALSE
+
+	if(!server)
+		return
+
+	var/mob/living/carbon/human/avatar/current_avatar = avatar_ref.resolve()
+	if(!current_avatar)
+		current_avatar = generate_avatar(neo)
+
+	if(!current_avatar)
+		balloon_alert(neo, "There is no room for you in the virtual domain.")
+		return
+
+	// Final check before we start the transfer
+	if(QDELETED(neo) || QDELETED(current_avatar) || neo.stat == DEAD || current_avatar.stat == DEAD)
+		return
+
+	var/datum/weakref/neo_ref = WEAKREF(neo)
+	voidrunner_ref = neo_ref
+	server.occupant_refs += neo_ref
+	avatar_ref = WEAKREF(current_avatar)
+	current_avatar.owner = neo
+
+	neo.mind.transfer_to(current_avatar, TRUE)
+	playsound(current_avatar, 'sound/magic/repulse.ogg', 30, 2)
+
+/// Generates a new avatar for the voidrunner.
+/obj/structure/netchair/proc/generate_avatar(mob/living/carbon/human/neo)
+	var/list/turf/possible_turfs = get_area_turfs(/area/station/virtual_domain/safehouse/exit, server.vdom.z_value)
+	if(!length(possible_turfs))
+		return FALSE
+
+	// Find an empty spot to spawn the ladder
+	var/turf/destination
+	for(var/turf/dest_turf as anything in possible_turfs)
+		if(!locate(/obj/structure/hololadder) in dest_turf)
+			destination = dest_turf
+			break
+	if(!destination)
+		return FALSE
+
+	var/obj/structure/hololadder/wayout = new(destination, src)
+	var/mob/living/carbon/human/avatar/avatar = new(wayout.loc, neo)
+	avatar.equipOutfit(netsuit, visualsOnly = TRUE)
+	return avatar
+
+/// Creates a list of outfit entries for the UI.
+/obj/structure/netchair/proc/make_outfit_collection(identifier, list/outfit_list)
+	var/list/collection = list(
+		"name" = identifier,
+		"outfits" = list()
+	)
+
+	for(var/path as anything in outfit_list)
+		var/datum/outfit/outfit = path
+		collection["outfits"] += list(list("path" = path, "name" = initial(outfit.name)))
+
+	return list(collection)
+
+/// On buckled, checks if mob is alive and human, then enters the matrix.
+/obj/structure/netchair/proc/on_buckle(datum/source, mob/living/mob_to_buckle, _force)
+	SIGNAL_HANDLER
+
+	if(!istype(mob_to_buckle) || !ishuman(mob_to_buckle) || mob_to_buckle.stat == DEAD)
+		return FALSE
+
+	enter_matrix(mob_to_buckle)
+
+/// Finds a quantum server to link to.
+/obj/structure/netchair/proc/panic_find_server()
+	for(var/obj/machinery/quantum_server/server as anything in oview(7))
+		if(istype(server, /obj/machinery/quantum_server))
+			src.server = server
+			return TRUE
+
+	return FALSE
+
+/// Resolves a path to an outfit.
+/obj/structure/netchair/proc/resolve_outfit(text)
+	var/path = text2path(text)
```

Comment:
This doesn't seem safe. I would use a list instead or check specifically if the outfit is in /outfit/job